Homepage nav bar

Elim-J opened the pull request.

Commit:
Front-end navbar+homepage

Still has a few bugs that need fixing. Updated homepage, navbar and router link to sign in page (can be edited to another page, just testing rn).

Used react routers for linking diff pages, and react scroll for linking to different sections on same page for navBar.

Need to do next:
Navbar highlights, images disappearing, sizing, margins, color corrections

Goals after bug fixes:
Add more pages and link them properly. Design new pages

Commit:
Updates

updates

Commit:
Fixed package.json for dependency issues

Commit:
Fixed package.json dependencies

Elim-J committed:
Fix bugs for homepage and navbar

Commit:
fixed one import

fixed an import and updated npm packages
pls run
npm install -g npm-check-updates
ncu -u
npm install
whenever using another branch for testing

Commit:
Merge branch 'Homepage-NavBar'

Commit:
Squashed commit of the following:

commit 398d30711584f3f355fdea2286dad3bbf77b125d
Merge: ff06333 e281dcc
Author: Vince Houghtaling <69610185+houghta4@users.noreply.github.com>
Date:   Tue Sep 20 22:12:31 2022 -0500

    Merge pull request #11 from Elim-J/sortingmodule

    clean up packages

commit e281dccfacafdb54f1e532c718f9c2542a8a9cf0
Author: ErikWenslaff <epw@uwm.edu>
Date:   Tue Sep 20 22:00:39 2022 -0500

    clean up packages

commit ff063338187ff510f356838f5d7defab5bf0222a
Merge: 6373490 e91bbf0
Author: ErikWenslaff <90141934+ErikWenslaff@users.noreply.github.com>
Date:   Tue Sep 20 21:32:58 2022 -0500

    Merge pull request #10 from Elim-J/sortingmodule

    create unit tests for bubblesort

commit e91bbf0d88079142027e47c0ada0593cee88af9d
Author: ErikWenslaff <epw@uwm.edu>
Date:   Tue Sep 20 21:22:53 2022 -0500

    create unit tests for bubblesort

commit 63734902543ac73772af7f5c25fbe18b26c7c52c
Merge: 7a93dc6 bd000d5
Author: ErikWenslaff <90141934+ErikWenslaff@users.noreply.github.com>
Date:   Sat Sep 17 19:16:41 2022 -0500

    Merge pull request #8 from Elim-J/sortingmodule

    Sortingmodule

commit bd000d52239db0bc229dc2629df4283b6b10ee7e
Author: ErikWenslaff <90141934+ErikWenslaff@users.noreply.github.com>
Date:   Sat Sep 17 19:00:06 2022 -0500

    moved to pathfindingmodule branch

commit 54aee16acfbc12f71dda1cd6d2de08e2448e0318
Author: ErikWenslaff <epw@uwm.edu>
Date:   Sat Sep 17 18:47:28 2022 -0500

    step buttons now pause and renamed 'skip to beginning' button

commit f62a030d9589599b4611970a426a75ca6aca121d
Author: ErikWenslaff <epw@uwm.edu>
Date:   Sat Sep 17 18:14:49 2022 -0500

    add bfs and bubblesort

Commit:
Merge branch 'Homepage-NavBar'

Commit:
Added link to algorithm page

need to style and add links to bubblesort from here, also need to implement a global navbar

Commit:
Adding bubblesort page to navbar from algorithm page

ErikWenslaffUWM committed:
create insertSort

Commit:
Update BubblesortPage.js

Commit:
Created quicksort.js

Completed quicksort.js, fixed all the bugs, generated frames and video.
All it needs is a UI

Commit:
updated packages

updated packages

houghta4 committed:
start of quicksort tree using d3-tree

Co-authored-by: ErikWenslaff <ErikWenslaff@users.noreply.github.com>

Elim-J committed:
Trello Nav Element

Add Trello to nav bar and create new Trello page

Elim-J committed:
Merge branch 'Homepage-NavBar' of https://github.com/Elim-J/capstone-project into Homepage-NavBar

ErikWenslaffUWM committed:
add fixTree and fixData methods

ErikWenslaffUWM committed:
added pivot position

pivots can be chosen by the user:
first
middle
last
random

Elim-J committed:
Trello

Create menu, create boards

houghta4 committed:
quicksort data displayed into d3 tree

Co-authored-by: ErikWenslaff <ErikWenslaff@users.noreply.github.com>

ErikWenslaffUWM committed:
fix generate random data

ErikWenslaffUWM committed:
added picklist for pivot selection

Elim-J committed:
trello updates

Add smaller menu for Delete and Edit

Elim-J committed:
Add additional dependencies

Commit:
Algo Landing Page

Added a carousel to algo landing page with cards including descriptions and a picture

Need to update links to each visualizer
need up update styling + pictures

Commit:
Merge branch 'Homepage-NavBar' of https://github.com/Elim-J/capstone-project into Homepage-NavBar

Commit:
Merge branch 'main' into Homepage-NavBar

WajahatAliShah1 reviewed (dismissed):
Updated the homepage-navbar branch with the main branch

WajahatAliShah1 reviewed (changes requested):
Fix Footer issues on pages, get better styling on algo landing page, populate router links to the proper urls for visualizers.

Also maybe get more trello functionality in, plus consider looking into styling updates for the rest of the pages: Algo landing, sorting, etc

ErikWenslaffUWM committed:
added insertsort.js

ErikWenslaffUWM committed:
Added Constants file for sorting algs

Commit:
Updated Links, cleaned up

Updated links to sorting and pathfinding interfaces. and Added the global sidebar to the other pages for a responsive navbar

ErikWenslaffUWM committed:
added picklist to select sorting algorithm

ErikWenslaffUWM committed:
fixed insertsort alg, visualization is now possible

ErikWenslaffUWM committed:
implemented randomsort

ErikWenslaffUWM committed:
implemented selection sort

Commit:
Merge branch 'sorting-buttons' into Homepage-NavBar

Commit:
Merge branch 'sortingmodule' into Homepage-NavBar

Commit:
Compiling sorting branches into main

Commit:
Update package-lock.json
